File: org.kevoree.modeling.genetic.framework/src/main/java/org/kevoree/modeling/genetic/api/FitnessFunction.java
package org.kevoree.modeling.genetic.api;

import org.kevoree.modeling.api.KMFContainer;

/**
 * Created with IntelliJ IDEA.
 * User: duke
 * Date: 12/02/13
 * Time: 14:40
 */
public interface FitnessFunction<A extends KMFContainer> {

    public double evaluate(A model);

}

File: org.kevoree.modeling.optimization.framework.core/src/main/java/org/kevoree/modeling/optimization/api/FitnessFunction.kt
package org.kevoree.modeling.optimization.api;

import org.kevoree.modeling.api.KMFContainer;
import org.kevoree.modeling.api.trace.TraceSequence

/**
 * Created with IntelliJ IDEA.
 * User: duke
 * Date: 12/02/13
 * Time: 14:40
 */
public trait FitnessFunction<A : KMFContainer> {

    public fun evaluate(model: A, origin: A?, traceSeq: TraceSequence?): Double ;

    public fun originAware() : Boolean;

}